src/main/java/teammates/sqllogic/core/UsersLogic.java

```diff
@@ -740,34 +746,49 @@ private boolean isSectionChanged(Section originalSection, Section newSection) {
     public Student updateStudentCascade(Student student)
             throws InvalidParametersException, EntityDoesNotExistException, EntityAlreadyExistsException {
 
-        Student originalStudent = getStudentForEmail(student.getCourseId(), student.getEmail());
+        String courseId = student.getCourseId();
+        Student originalStudent = getStudent(student.getId());
+        String originalEmail = originalStudent.getEmail();
+        boolean changedEmail = isEmailChanged(originalEmail, student.getEmail());
+
+        // check for email conflict
+        Student s = usersDb.getStudentForEmail(courseId, student.getEmail());
+        if (changedEmail && s != null) {
+            String errorMessage = String.format(ERROR_CREATE_ENTITY_ALREADY_EXISTS, s.toString());
+            throw new EntityAlreadyExistsException(errorMessage);
+        }
+
         Team originalTeam = originalStudent.getTeam();
         Section originalSection = originalStudent.getSection();
-
         boolean changedTeam = isTeamChanged(originalTeam, student.getTeam());
         boolean changedSection = isSectionChanged(originalSection, student.getSection());
 
+        // update student
+        usersDb.checkBeforeUpdateStudent(student);
         originalStudent.setName(student.getName());
         originalStudent.setTeam(student.getTeam());
         originalStudent.setEmail(student.getEmail());
         originalStudent.setComments(student.getComments());
 
-        Student updatedStudent = usersDb.updateStudent(originalStudent);
```

Comment:
Was this removed by mistake?

Reply:
I think Hibernate automatically persists the changes, as per the [documentation](https://docs.jboss.org/hibernate/orm/4.0/devguide/en-US/html_single/#d0e1739).

This test in `UpdateStudentActionIT` should also verify that the changes are being persisted:
```
public void testExecute_typicalCase_success() throws Exception {
...
        Student updatedStudent = logic.getStudent(student1.getId());
        assertEquals(updatedStudent.getEmail(), newStudentEmail);
        assertEquals(updatedStudent.getTeamName(), newStudentTeam);
        assertEquals(updatedStudent.getComments(), newStudentComments);
```